config/swagger.yaml

```diff
@@ -503,9 +503,289 @@ paths:
+          type: boolean
+          default: false
+      responses:
+        '200':
+          description: The block being found 
+          schema:
+            $ref: '#/definitions/GenericBlock'
+        '404':
+          description: Block not found
+          schema:
+            $ref: '#/definitions/Error'
+      security: []
+  /block/hash/{hash}:
+    get:
+      tags:
+        - internal
+      operationId: GetBlockByHashInternal
+      description: 'Get a block by hash'
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters: 
+        - in: path 
+          name: hash
+          description: 'Hash of the block to fetch'
+          required: true
+          type : string 
+        - in: query
+          name: tx_objects
+          description: 'Transactions as objects (default is as hashes)'
+          type: boolean
+          default: false
+      responses:
+        '200':
+          description: The block being found  
+          schema:
+            $ref: '#/definitions/GenericBlock'
+        '400':
+          description: Invalid hash 
+          schema:
+            $ref: '#/definitions/Error'
+        '404':
+          description: Block not found
+          schema:
+            $ref: '#/definitions/Error'
+      security: []
+  /block/genesis:
+    get:
+      tags:
+        - internal
+      operationId: GetBlockGenesis
+      description: 'Get the genesis block'
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: tx_objects
+          description: 'Transactions as objects (default is as hashes)'
+          type: boolean
+          default: false
+      responses:
+        '200':
+          description: The genesis block 
+          schema:
+            $ref: '#/definitions/GenericBlock'
+      security: []
+  /block/latest:
+    get:
+      tags:
+        - internal
+      operationId: GetBlockLatest
+      description: 'Get the top block'
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: tx_objects
+          description: 'Transactions as objects (default is as hashes)'
+          type: boolean
+          default: false
+      responses:
+        '200':
+          description: The top block 
+          schema:
+            $ref: '#/definitions/GenericBlock'
+      security: []
+  /block/pending:
+    get:
+      tags:
+        - internal
+      operationId: GetBlockPending
+      description: 'Get the block being mined'
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - in: query
+          name: tx_objects
+          description: 'Transactions as objects (default is as hashes)'
+          type: boolean
+          default: false
+      responses:
+        '200':
+          description: The pending block 
```

Comment:
Please mention that nonce and Cuckoo solution evidence are not meaningful (distinct type would be cleaner but too complex). (If only one miner as now, actually there is a nonce identified.)

Reply:
I see amended - thanks.